x/group/types/conditions.go

```diff
@@ -0,0 +1,132 @@
+package types
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/regen-network/regen-ledger/orm"
+)
+
+var (
+	// AddressLength is the length of all addresses
+	// You can modify it in init() before any addresses are calculated,
+	// but it must not change during the lifetime of the kvstore
+	AddressLength = 20
+
+	// it must have (?s) flags, otherwise it errors when last section contains 0x20 (newline)
```

Comment:
regular expression is not a function it doesn't return an error. 
1. What does `perm` mean?
2. let's update a comment and explain what we want to find or match.

Reply:
OK, I found later that `perm` is permission. Let's describe it in a comment.